Read and create library files in the directory given by MtxLibDir.

The patch keeps a promise given by upstream.

AUTHOR: Simon King 2015-09-18, simon.king@uni-jena.de

diff --git a/src/maketabF.c b/src/maketabF.c
index fa03eda..d7af83e 100644
--- a/src/maketabF.c
+++ b/src/maketabF.c
@@ -319,7 +319,7 @@ static void writeheader()
     int i, j;
 
     sprintf(filename,"p%3.3ld.zzz",Q);
-    fd = SysFopen(filename,FM_CREATE);
+    fd = SysFopen(filename,FM_CREATE|FM_LIB);
     if (fd == NULL)
     {
 	perror(filename);
diff --git a/src/os.c b/src/os.c
index a7f4271..b07b971 100644
--- a/src/os.c
+++ b/src/os.c
@@ -227,25 +227,31 @@ FILE *SysFopen(const char *name, int mode)
 	MTX_ERROR1("Invalid file mode %d",mode);
 	return NULL;
     }
-    f = fopen(name,fmodes[m]);
-    if (f != NULL) 
-	return f;
 
     /* Search library directory
        ------------------------ */
     if ((mode & FM_LIB) != 0) 
     {
-	strcpy(buf,MtxLibDir);
-	strcat(buf,"/");
-	strcat(buf,name);
-	f = fopen(buf,fmodes[m]);
+        if (*MtxLibDir != 0)
+            {
+            strcpy(buf,MtxLibDir);
+            strcat(buf,"/");
+            strcat(buf,name);
+            f = fopen(buf,fmodes[m]);
+            }
+        else
+            f = fopen(name,fmodes[m]);
     }
-
+    else
+    {
+        f = fopen(name,fmodes[m]);
+    }
+    if (f != NULL)
+	return f;
     /* Error handling
        -------------- */
     if (f == NULL && (mode & FM_NOERROR) == 0)
-	MTX_ERROR1("%s: %S",name);
-
+    MTX_ERROR1("%s: %S",name);
     return f;
 }
 
diff --git a/src/zcv.c b/src/zcv.c
index a9ad7a3..763c9fb 100644
--- a/src/zcv.c
+++ b/src/zcv.c
@@ -584,7 +584,7 @@ static int Init(int argc, const char **argv)
     inpname = App->ArgV[0];
     if (strcmp(inpname,"-"))
     {
-	src = SysFopen(inpname,FM_READ|FM_TEXT|FM_LIB);
+	src = SysFopen(inpname,FM_READ|FM_TEXT);
 	if (src == NULL)
 	{
 	    MTX_ERROR1("Cannot open %s",inpname);
